<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Cultured Genome Assembly | Aaron’s Bioinformatics Notes</title>
  <meta name="description" content="Some snapshots of software and commands I use to run various metagenomic analyses." />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Cultured Genome Assembly | Aaron’s Bioinformatics Notes" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Some snapshots of software and commands I use to run various metagenomic analyses." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Cultured Genome Assembly | Aaron’s Bioinformatics Notes" />
  
  <meta name="twitter:description" content="Some snapshots of software and commands I use to run various metagenomic analyses." />
  

<meta name="author" content="Aaron Oliver" />


<meta name="date" content="2023-07-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="genome-annotation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Aaron's Bioinformatics Notes</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a></li>
<li class="chapter" data-level="2" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html"><i class="fa fa-check"></i><b>2</b> Cultured Genome Assembly</a>
<ul>
<li class="chapter" data-level="2.1" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#short-read-assembly"><i class="fa fa-check"></i><b>2.1</b> Short read assembly</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#quality-control"><i class="fa fa-check"></i><b>2.1.1</b> Quality control</a></li>
<li class="chapter" data-level="2.1.2" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#trimming"><i class="fa fa-check"></i><b>2.1.2</b> Trimming</a></li>
<li class="chapter" data-level="2.1.3" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#running-the-assembler"><i class="fa fa-check"></i><b>2.1.3</b> Running the assembler</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#long-read-assembly"><i class="fa fa-check"></i><b>2.2</b> Long read assembly</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#quality-control-1"><i class="fa fa-check"></i><b>2.2.1</b> Quality control</a></li>
<li class="chapter" data-level="2.2.2" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#running-the-assembler-1"><i class="fa fa-check"></i><b>2.2.2</b> Running the assembler</a></li>
<li class="chapter" data-level="2.2.3" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#polishing-with-medaka-ont"><i class="fa fa-check"></i><b>2.2.3</b> Polishing with Medaka (ONT)</a></li>
<li class="chapter" data-level="2.2.4" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#polishing-with-racon-pacbio"><i class="fa fa-check"></i><b>2.2.4</b> Polishing with Racon (PacBio)</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#assembly-assessment"><i class="fa fa-check"></i><b>2.3</b> Assembly assessment</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#prokaryotic-genome"><i class="fa fa-check"></i><b>2.3.1</b> Prokaryotic genome</a></li>
<li class="chapter" data-level="2.3.2" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#eukaryotic-genome"><i class="fa fa-check"></i><b>2.3.2</b> Eukaryotic genome</a></li>
<li class="chapter" data-level="2.3.3" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#viral-genome"><i class="fa fa-check"></i><b>2.3.3</b> Viral genome</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="cultured-genome-assembly.html"><a href="cultured-genome-assembly.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genome-annotation.html"><a href="genome-annotation.html"><i class="fa fa-check"></i><b>3</b> Genome Annotation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genome-annotation.html"><a href="genome-annotation.html#building-gene-models"><i class="fa fa-check"></i><b>3.1</b> Building gene models</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="genome-annotation.html"><a href="genome-annotation.html#prokaryotes"><i class="fa fa-check"></i><b>3.1.1</b> Prokaryotes</a></li>
<li class="chapter" data-level="3.1.2" data-path="genome-annotation.html"><a href="genome-annotation.html#eukaryotes"><i class="fa fa-check"></i><b>3.1.2</b> Eukaryotes</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="genome-annotation.html"><a href="genome-annotation.html#annotating-proteins"><i class="fa fa-check"></i><b>3.2</b> Annotating proteins</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="metaG.html"><a href="metaG.html"><i class="fa fa-check"></i><b>4</b> Metagenomic Assembly and Binning</a>
<ul>
<li class="chapter" data-level="4.1" data-path="metaG.html"><a href="metaG.html#exploring-the-bins"><i class="fa fa-check"></i><b>4.1</b> Exploring the bins</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="metaG.html"><a href="metaG.html#bin-completeness-and-quality-metrics"><i class="fa fa-check"></i><b>4.1.1</b> Bin completeness and quality metrics</a></li>
<li class="chapter" data-level="4.1.2" data-path="metaG.html"><a href="metaG.html#assess-genome-taxonomy"><i class="fa fa-check"></i><b>4.1.2</b> Assess genome taxonomy</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">A bookdown publication</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Aaron’s Bioinformatics Notes</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cultured-genome-assembly" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Cultured Genome Assembly<a href="cultured-genome-assembly.html#cultured-genome-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>talk about the differences between short and long read assembly</p>
<div id="short-read-assembly" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Short read assembly<a href="cultured-genome-assembly.html#short-read-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="quality-control" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Quality control<a href="cultured-genome-assembly.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="trimming" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Trimming<a href="cultured-genome-assembly.html#trimming" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="running-the-assembler" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Running the assembler<a href="cultured-genome-assembly.html#running-the-assembler" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="cultured-genome-assembly.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> assembly/assembly.fasta best_assembly.fasta</span></code></pre></div>
</div>
</div>
<div id="long-read-assembly" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Long read assembly<a href="cultured-genome-assembly.html#long-read-assembly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Long read (PacBio and ONT) assembly uses a different suite of tools due to differences in assembly methods and datatypes.</p>
<div id="quality-control-1" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Quality control<a href="cultured-genome-assembly.html#quality-control-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>LongQC [<a href="https://github.com/yfukasawa/LongQC">GitHub</a>, <a href="https://anaconda.org/bioconda/longqc">Conda</a>, <a href="https://doi.org/10.1534/g3.119.400864">Paper</a>]</p>
<p>Long re</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="cultured-genome-assembly.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">python</span> longQC.py sampleqc <span class="at">-x</span> [sequencer] <span class="at">-o</span> qc_output raw_reads.fq</span></code></pre></div>
<p>where the sequencer can be one of the following: <code>pb-rs2</code>, <code>pb-sequel</code>, <code>pb-hifi</code>, <code>ont-ligation</code> (PromethION), or <code>ont-rapid</code> (MinION, GridION, Flongle).</p>
</div>
<div id="running-the-assembler-1" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Running the assembler<a href="cultured-genome-assembly.html#running-the-assembler-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are many valid choices of assembler for long reads. Among the most popular and best validated are Flye (ONT/PacBio), Canu (ONT/PacBio), miniasm (ONT/PacBio), Raven (ONT/PacBio), Falcon (PacBio), and the hybrid (Illumina + ONT/PacBio) assemblers Unicycler and hybridSPAdes.</p>
<p>Each assembler comes with different usecases and drawbacks. For example, Canu is very slow and Falcon can only use PacBio reads, but both are diploid-aware assemblers which might be critical for certain use cases. Hybrid assembly can work well when single nucleotide differences are important, such as when a strain-level reference genome is available.</p>
<p>I will use Flye [<a href="https://github.com/fenderglass/Flye">GitHub</a>, <a href="https://anaconda.org/bioconda/flye">Conda</a>, <a href="https://doi.org/10.1038/s41587-019-0072-8">Paper</a>]
for this tutorial, since it also has a metagenomic mode for long read MAG binning. So once you are comfortable using Flye for genomes, you can use it for metagenomes too!</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="cultured-genome-assembly.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">flye</span> [sequence-type] <span class="at">--genome-size</span> [] raw_reads.fq <span class="at">--out-dir</span> assembly</span></code></pre></div>
<p>where the sequence type is one of <code>--pacbio-raw</code>, <code>--pacbio-hifi</code>, <code>--nano-raw</code>, or <code>--nano-hq</code>. Estimating genome size based on a quick skim of closest relatives can give you a better assembly, but its optional. The estimated genome size can be formatted in Megabases or Gigabases, e.g. 200m or 3.6g. The CPU parameter for flye is <code>--threads</code>.</p>
<p>The output assembly will be located in <code>assembly/assembly.fasta</code>.</p>
</div>
<div id="polishing-with-medaka-ont" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Polishing with Medaka (ONT)<a href="cultured-genome-assembly.html#polishing-with-medaka-ont" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are also many polishers to be used with long read data. For ONT sequencing, Oxford Nanopore’s own
Medaka polisher [<a href="https://github.com/nanoporetech/medaka">GitHub</a>, <a href="https://anaconda.org/bioconda/medaka">Conda</a>] works well, is very fast, and can polish using the same reads input into the assembly. Since Medaka is specifically built to work best with Flye, we will use it for this tutorial.</p>
<p>To run Medaka:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="cultured-genome-assembly.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">medaka_consensus</span> <span class="at">-i</span> raw_reads.fq <span class="at">-d</span> assembly/assembly.fasta <span class="at">-o</span> medaka_polished <span class="at">-m</span> [model]</span></code></pre></div>
<p><a href="https://github.com/nanoporetech/medaka#models">See here for more information about how to choose the appropriate model</a>. Your results will be better if you carefully choose the right model, but the program should also work if you just provide <code>r941_min_fast_g303</code> for MinION/GridION or <code>r941_prom_hac_g303</code> for high accuracy PrometION data. The CPU parameter for medaka is <code>-t</code>.</p>
<p>Your new polished assembly will be located in <code>medaka_polished/consensus.fasta</code>.</p>
<p>For consistency with the short read tutorial for the steps below, I am going to create a symbolic link to our best assembly that we will use for the assembly assessment steps.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="cultured-genome-assembly.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> medaka_polished/consensus.fasta best_assembly.fasta</span></code></pre></div>
</div>
<div id="polishing-with-racon-pacbio" class="section level3 hasAnchor" number="2.2.4">
<h3><span class="header-section-number">2.2.4</span> Polishing with Racon (PacBio)<a href="cultured-genome-assembly.html#polishing-with-racon-pacbio" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For PacBio data, I suggest polishing with Racon [<a href="https://github.com/lbcb-sci/racon">GitHub</a>, <a href="https://anaconda.org/bioconda/racon">Conda</a>, <a href="https://doi.org/10.1101/gr.214270.116">Paper</a>].</p>
</div>
</div>
<div id="assembly-assessment" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Assembly assessment<a href="cultured-genome-assembly.html#assembly-assessment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>QUAST [<a href="https://github.com/ablab/quast">GitHub</a>, <a href="https://anaconda.org/bioconda/quast">Conda</a>] will give</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="cultured-genome-assembly.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">quast.py</span> best_assembly.fasta <span class="at">-o</span> quast_output</span></code></pre></div>
<p>This command will provide a set of summary files that you can use to evaluate your assembly.
In particular, the N50, number of contigs, total length of all contigs, and the length of the largest contig are all important metrics that are often reported with genome assemblies. The N50 is <a href="https://en.wikipedia.org/wiki/N50,_L50,_and_related_statistics">the length of a contig, such that all the contigs of at least the same length together cover at least 50% of the assembly</a>.</p>
<p>While assembly metrics are important, what you will more often be interested in is the level of completeness and contamination in your assembled genome. Which tool you use will depend on what kind of genome you are assembling.
The final output of these tools provide completeness (what percentage of expected marker genes were detected) and contamination (what percentage of expected marker genes were detected more than once).</p>
<p>A threshold of 50% completeness and 10% contamination is considered a medium-quality genome, and 90% completeness and 5% contamination is necessary (but not sufficient) to have a high-quality genome.</p>
<div id="prokaryotic-genome" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Prokaryotic genome<a href="cultured-genome-assembly.html#prokaryotic-genome" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>CheckM used to be the gold standard for bacterial completeness analysis. After nine years, CheckM2 has released in 2023, so we now have a new state of the art.</p>
<p>CheckM2 has a simple command to generate completeness metrics for you.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="cultured-genome-assembly.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">checkm2</span> predict <span class="at">--input</span> best_assembly.fasta <span class="at">--output-directory</span> checkm2</span></code></pre></div>
<p>CheckM will automatically determine the lineage of our genome and then select a suite of single-copy marker genes relevant to that taxonomic group. <code>--threads</code> is the parameter for CPUs.</p>
</div>
<div id="eukaryotic-genome" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Eukaryotic genome<a href="cultured-genome-assembly.html#eukaryotic-genome" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>BUSCO is an alternative completeness analysis tool with the added benefit of working with both prokaryotic and eukaryotic lineages. In general, I recommend using BUSCO primarily for eukaryotic assemblies, and only using it for prokaryotic assemblies as a point of comparison to CheckM2.</p>
<p>To use BUSCO, you will first have to find the lineage closest to your genome in the <a href="https://busco.ezlab.org/list_of_lineages.html">list of available BUSCO lineages</a>. For example, if you had a diatom genome, the best lineage would be <code>stramenopiles_odb10</code>. If you are unsure of your genome’s lineage, you can run BUSCO’s automated lineage selection by omitting the <code>-l</code> parameter, but it is slow and does not work as well with eukaryotes.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="cultured-genome-assembly.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">busco</span> <span class="at">-i</span> best_assembly.fasta <span class="at">-l</span> [LINEAGE] <span class="at">-o</span> busco <span class="at">-m</span> [MODE]</span></code></pre></div>
<p>The mode parameter <code>-m</code> should be one of <code>genome</code>, <code>proteins</code>, or <code>transcriptome</code>, depending on your data set.</p>
<p>The output file <code>busco/short_summary.txt</code> will provide completeness and contamination metrics.</p>
</div>
<div id="viral-genome" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Viral genome<a href="cultured-genome-assembly.html#viral-genome" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You may still want to do this step, even if you didn’t set out to assemble a viral genome! There is already a nice protocol/tutorial that I use for this step, <a href="https://www.protocols.io/view/viral-sequence-identification-sop-with-virsorter2-5qpvoyqebg4o/v3">so I’m just going to link it here!</a></p>
</div>
</div>
<div id="summary" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Summary<a href="cultured-genome-assembly.html#summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Here is the data you have generated up until this point:</p>
<ul>
<li>Quality control metrics on your reads</li>
<li>A taxonomic distribution of your reads</li>
<li>Cleaned reads with adapters and low quality sequences removed</li>
<li>An assembled genome</li>
<li>Assembly statistics describing how well your genome assembled</li>
<li>Completeness and contamination statistics for your genome</li>
</ul>
<p>Now, you’ll need to annotate your genome to understand its metabolic capacities, perform phylogenetics, and detect novel proteins. Lucky for you, such a tutorial is available in the next chapter!</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="genome-annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
